Sources/Web3Core/Contract/ContractProtocol.swift

```diff
@@ -121,7 +121,7 @@ public protocol ContractProtocol {
     /// - Returns: Encoded data for a given parameters, which is should be assigned to ``CodableTransaction.data`` property
     func deploy(bytecode: Data,
                 constructor: ABI.Element.Constructor?,
-                parameters: [AnyObject]?,
+                parameters: [Any]?,
```

Comment:
I see that in some places you drop a type erasure totally, while in others you've left it, but change the target type to `Any`? Within my review I assume that you do this in following logic: in case if type erasure it was removed, in case its required target type has ben set to `Any`. Tell me if i'm wrong with that, resolve this otherwise.

Reply:
Could you please provide one or more examples where I did the following:
> in some places you drop a type erasure totally

I'll start to self-review this PR only now.

Reply:
To remove any misunderstanding: the goal I had in mind is to eliminate the use `AnyObject` as much as possible. I simply performed the replacement of `AnyObject` with `Any`, ran the tests, fixed almost all issues that occurred (except ENS tests) and opened this PR.

Reply:
`AnyObject` uses bridged types and basically changes the type of the object when it can.
For example:
- `String` -> `NSString`;
- `Int` -> `NSNumber`;
- `Array` -> `NSArray`;
- `struct` -> `__SwiftValue`;
- etc.

While using `Any` we preserve this information. 

<img width="786" alt="Screenshot 2023-01-13 at 21 09 16" src="https://user-images.githubusercontent.com/36865532/212399477-caa405c5-17ec-49ff-bf2a-cb62c304e45a.png">


<img width="453" alt="Screenshot 2023-01-13 at 21 04 38" src="https://user-images.githubusercontent.com/36865532/212399354-b97a27ee-7f3e-4c20-802b-c475e7437c6f.png">


And this is where it gets important!
When we have `Int8`, `Int16`, `Int..` and `UInt[8-64]` these all get represented as `NSNumber` if we cast them to `AnyObject`. And here is an example of what will happen:
<img width="489" alt="Screenshot 2023-01-13 at 21 19 13" src="https://user-images.githubusercontent.com/36865532/212401404-faa625c4-bfa0-46ae-b0e3-be1d6c21d91d.png">

<img width="300" alt="Screenshot 2023-01-13 at 21 19 29" src="https://user-images.githubusercontent.com/36865532/212401433-1dab1345-8c7f-4ff0-acb4-1cb839097b0f.png">

We do not want to represent `Int8` as `Int` or the other way around.

Reply:
I guess I've finally got the point. AnyObject is about bridging swift types to an Obj-C one, but with type erasure. Guess it's some internal apple demand at its most.